src/bitcoin-cli.cpp

```diff
@@ -298,30 +296,22 @@ class GetinfoRequestHandler: public BaseRequestHandler
 class NetinfoRequestHandler : public BaseRequestHandler
 {
 private:
-    bool IsAddrIPv6(const std::string& addr) const
+    const std::vector<std::pair<std::string, uint8_t>> m_networks{{"ipv4", 0}, {"ipv6", 1}, {"onion", 2}};
+    const size_t m_networks_size{m_networks.size()};
+    uint8_t NetworkStringToId(const std::string& str) const
     {
-        return !addr.empty() && addr.front() == '[';
-    }
-    bool IsInboundOnion(const std::string& addr_local, int mapped_as) const
-    {
-        return mapped_as == 0 && addr_local.find(ONION) != std::string::npos;
-    }
-    bool IsOutboundOnion(const std::string& addr, int mapped_as) const
-    {
-        const size_t addr_len{addr.size()};
-        const size_t onion_pos{addr.rfind(ONION)};
-        return mapped_as == 0 && onion_pos != std::string::npos && addr_len > ONION_LEN &&
-               (onion_pos == addr_len - ONION_LEN || onion_pos == addr.find_last_of(":") - ONION_LEN);
+        for (const auto& network : m_networks) {
+            if (str == network.first) return network.second;
+        }
+        assert(false);
```

Comment:
Let's not do `assert(false)` here. A crash on "invalid" import from the RPC is never good, it is good to be slightly forgiving with regard to forward-compatibility. Maybe return an 'unknown' category instead that is not used for the statistics/table.

Reply:
Done!